modules/cas_disk/exp_obj.c

```diff
@@ -339,8 +339,40 @@ static void _casdsk_exp_obj_clear_dev_t(struct casdsk_disk *dsk)
 	}
 }
 
+static int _casdsk_exp_obj_open(struct block_device *bdev, fmode_t mode)
+{
+	struct casdsk_disk *dsk = bdev->bd_disk->private_data;
+	int result;
```

Comment:
How about initializing `result` with `-ENAVAIL`? This way you could get rid of the first branch of `if`